ops/services/app_service/main.tf

```diff
@@ -164,10 +164,12 @@ resource "azurerm_app_service_slot_virtual_network_swift_connection" "staging" {
   WHAT'S HAPPENING HERE:
 
   1) The wildcard-simplereport-gov cert is being imported from Key Vault into the App Service
+    [NOTE: This only takes place if this is the first environment being created in this environment level. Cert ownership is bound to the index-less environment!]
   2) That cert is being bound to the custom domain created above, enabling HTTPS
 */
 
 resource "azurerm_app_service_certificate" "app" {
+  count               = var.env_index == 1 ? 1 : 0
```

Comment:
Here's the catch with this...I tried to have this create every time, but Azure/TF yells and blocks the operation. At first, I thought it was a name issue, so I parameterized the name, but it _still_ blocked, because it only allows for a single instance of a certificate fingerprint in a specific resource group. The custom hostname bindings that are created in the block below can each leverage the same cert, which is why I had to do this weird bit of logic that I still honestly don't like. 😥 

I played with this for a few hours last night, and this was the most elegant thing I could think of that would let us work within these constraints. I'm certainly all ears for something less kludgy than this, though!

Reply:
Ah, that does change things then. I think this is fine in that case. It will change the state name (to `azurerm_app_service_certificate.app[0]`) but I think the delete/recreate it will cause is fine (Terraform will handle re-binding it) and there's no need to rename the state ahead of time.